### DISCUSSION TITLE
想判断 是谁给机器人 发的消息，不同的人执行不同命令 在哪里 写判断

### DISCUSSION THREAD

**Discussion by yixianbaowang** [MR-XieXuan/MiraiTravel#8] [Q&A] [answered]
想判断 是谁给机器人 发的消息，不同的人执行不同命令 在哪里 写判断

**Comment by MR-XieXuan** [comment, marked as answer]
你在查看webhook日志时,你就可以看到 你的QQBot 接收到了一串消息,这个消息实际上就是 Mirai 的webhook上报机制上报的消息
![image](https://user-images.githubusercontent.com/95032548/218393910-58e97c5e-521f-4f0d-a470-a30a6425a51d.png)
根据消息类型的不同,我们可以查看Mirai的文档进行判断该消息的类型等,我们以群消息为例。在Mirai文档中如下：https://docs.mirai.mamoe.net/mirai-api-http/api/MessageType.html#%E5%A5%BD%E5%8F%8B%E6%B6%88%E6%81%AF
![image](https://user-images.githubusercontent.com/95032548/218394251-7ae050fd-970e-4207-8f8f-698dd0f3c938.png)
我们可以看到发送者的信息被存放在sender中,于是我们获取到这个内容。
而考虑到我们一般采用miraiTravel的webhook组件来对QQ机器人进行控制。所以我说一下在webhook组件中webhook_group_message获取该信息的方法:
该信息会当作第一个参数参数传入QQObj的 webhook_xxx 中,类型是 list 。
所以我们获取发送消息的人的QQ号的方法为 :
```php
webhook_group_message( $e ){
    reply_message(  "你的QQ号是:" .  $e['sender']['id']  , true ) ;
}
```
同理获取发送消息的人和群的 群号码为 : 
```php
webhook_group_message($e){
    reply_message( " 群号为 ： "  . $e['sender']['group']['id'] , true);
}
```

**Reply by MR-XieXuan** [reply]
@yixianbaowang  还是得多看手册啊，兄弟。

**Reply by MR-XieXuan** [reply]
在哪里写判断的话只要能跑的地方写判断即可。
